examples/migration/CreateCompleteCampaignAdWordsApiOnly.py

```diff
@@ -0,0 +1,202 @@
+KEYWORDS_TO_ADD = ["mars cruise", "space hotel" ]
+import urllib.parse
+
+
+def createCampaignBudget(client):
+   # Create a budget, which can be shared by multiple campaigns.
+    budget_service = client.GetService('BudgetService', version='v201809')
+    budget = {
+        'name' : 'Interplanetary Cruise Budget #%s' % uuid.uuid4(),
+        'amount': {
+            'microAmount' : '50000000'
+        },
+        'deliveryMethod' : 'STANDARD'
+    }
+    budget_operations = [{
+        'operator': 'ADD',
+        'operand': budget
+    }]
+    # Add budget.
+    results = budget_service.mutate(budget_operations)
+    createdBudget = results['value'][0]
+    print("Budget with ID {} and name {} was created".format(\
+           createdBudget['budgetId'],createdBudget['name']))
+    return createdBudget['budgetId']
+
+
+def createCampaign(client, budgetId):
+    campaign_service = client.GetService('CampaignService', version='v201809')
+    campaign = {
+        'name': 'Interplanetary Cruise #%s' % uuid.uuid4(),
+        'advertisingChannelType': 'SEARCH',
+        # Recommendation: Set the campaign to PAUSED when creating it to stop the
+        # ads from immediately serving. Set to ENABLED once you've added
+        # targeting and the ads are ready to serve.
+        'status': 'PAUSED',
+        'biddingStrategyConfiguration': {
+            'biddingStrategyType': 'MANUAL_CPC',
+        },
+        'startDate': (datetime.datetime.now() +
+        datetime.timedelta(1)).strftime('%Y%m%d'),
+        'endDate': (datetime.datetime.now() +
+        datetime.timedelta(365)).strftime('%Y%m%d'),
+        # Budget (required) - note only the budget ID is required.
+        'budget': {
+            'budgetId': budgetId
+        },
+        'networkSetting': {
+            'targetGoogleSearch': 'true',
+            'targetSearchNetwork': 'true',
+        }
+    }
+    campaign_operations = [{
+        'operator': 'ADD',
+        'operand': campaign
+    }]
+    results = campaign_service.mutate(campaign_operations)
+    createdCampaign = results['value'][0]
+    print("CreatedCampign with ID {} and name {} was created". \
+           format(createdCampaign['id'],createdCampaign['name']))
+    return createdCampaign['id']
+
+
+def createAdGroup(client, campaign_id):
+    ad_group_service = client.GetService('AdGroupService', 'v201809')
+    ad_group =   {
+        'name': 'Earth to Mars Cruise #%s' % uuid.uuid4(),
+        'campaignId': campaign_id,
+        'status': 'ENABLED',
+        'biddingStrategyConfiguration' : {
+        'bids': [{
+          # The 'xsi_type' field allows you to specify the xsi:type of the
+          # object being created. It's only necessary when you must provide
+          # an explicit type that the client library can't infer.
+            'xsi_type': 'CpcBid',
+            'bid': {
+                'microAmount': 10000000
+            }
+        }]
+        },
+        'adGroupAdRotationMode': 'OPTIMIZE'
```

Comment:
The above two lines should be indented 4 spaces not 8.

Reply:
I am not sure about this. Shouldn't the previous two brackets end with 8 spaces, as they are directly below the 'bid' variable.

Reply:
Oh right I misread, actually `'bids'` (lines 104 - 112) should be indented another 4 spaces since they're inside the `'biddingStrategyConfiguration'` block.